src/Nest/IndexModules/IndexSettings/Settings/IndexSettingsConverter.cs

```diff
@@ -20,76 +21,81 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
 			if (ds == null) return;
 			IDictionary<string,object> d = ds;
 
-			d[UpdatableIndexSettings.NumberOfReplicas] = ds.NumberOfReplicas;
-			d[UpdatableIndexSettings.AutoExpandReplicas] = ds.AutoExpandReplicas;
-			d[UpdatableIndexSettings.RefreshInterval] = ds.RefreshInterval;
-			d[UpdatableIndexSettings.BlocksReadOnly] = ds.BlocksReadOnly;
-			d[UpdatableIndexSettings.BlocksRead] = ds.BlocksRead;
-			d[UpdatableIndexSettings.BlocksWrite] = ds.BlocksWrite;
-			d[UpdatableIndexSettings.BlocksMetadata] = ds.BlocksMetadata;
-			d[UpdatableIndexSettings.Priority] = ds.Priority;
-			d[UpdatableIndexSettings.RecoveryInitialShards] = ds.RecoveryInitialShards;
-			d[UpdatableIndexSettings.RequestsCacheEnable] = ds.RequestsCacheEnabled;
-			d[UpdatableIndexSettings.RoutingAllocationTotalShardsPerNode] = ds.RoutingAllocationTotalShardsPerNode;
-			d[UpdatableIndexSettings.UnassignedNodeLeftDelayedTimeout] = ds.UnassignedNodeLeftDelayedTimeout;
+			void Set(string knownKey, object newValue)
+			{
+				if (newValue != null) d[knownKey] = newValue;
```

Comment:
Might you want to set a null value too?

Reply:
That would always override what is set in the backing dictionary which is what triggered #3019